controllers/storagecluster/storagecluster_controller.go

```diff
@@ -233,12 +233,26 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&extv1.CustomResourceDefinition{},
 			enqueueStorageClusterRequest,
 			builder.WithPredicates(
-				util.NamePredicate(VirtualMachineCrdName),
-				util.CrdCreateAndDeletePredicate(&r.Log, VirtualMachineCrdName, r.AvailableCrds[VirtualMachineCrdName]),
-			),
-			builder.WithPredicates(
-				util.NamePredicate(StorageClientCrdName),
-				util.CrdCreateAndDeletePredicate(&r.Log, StorageClientCrdName, r.AvailableCrds[StorageClientCrdName]),
+				predicate.Or(
```

Comment:
Not able to understand the calling pattern of this function. Can you please add few lines of comment here to explain what exactly we are trying to achieve here, will be helpful for future changes as well

Reply:
> Not able to understand the calling pattern of this function
- may I know if the issue was understood based on the PR description? let me know if below makes sense and I'll add the same as comment.

requirement:
1. if the required CRDs state changes we want to reconcile and the interested state is if there are created or deleted
2. the function accepts booleans corresponding to create, update, delete and generic. we always supply false to update & generic as we don't need reconcile for that
3. now we want to get reconcile if
a. if crd didn't exist at start of controller and now it became availabe i.e, options for create event
b. if crd exist at start of controller and now it is deleted ie., option for delete event
4. point 3 is for a specific crd and we want the same for multiple CRDs and so the condition is `OR( (name AND event) ... )`

Reply:
Thanks for the detailed explanation, I understand now.